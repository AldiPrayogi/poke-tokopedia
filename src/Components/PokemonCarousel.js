/** @jsxImportSource @emotion/react */

import React from "react";
import {Carousel, WhiteSpace, WingBlank} from "antd-mobile";
import {CarouselComponentCSS} from '../Styling/ComponentCSS'
import {useHistory} from 'react-router-dom';

export const PokemonCarousel = (data) => {
    const history = useHistory();

    const handlePokemonClick =  (name) => {
        history.push({
            pathname: `/pokemon-detail/${name}`
        });
    }

    return(
        <div css={CarouselComponentCSS}>
            <WingBlank>
                <Carousel
                    autoplay={false}
                    infinite
                    cellSpacing={10}
                    slideWidth={1}
                >
                    {data.data.map(val => (
                        <div className='carousel-container'>
                            <div className='pokemon-name-header'>
                                <h4 className='pokemon-name'>{(val.name).toUpperCase()}</h4>
                            </div>
                            <WhiteSpace size='xl'/>
                            <img
                                className='pokemon-image'
                                src={val.image}
                                alt={val.name}
                                onClick={() => handlePokemonClick(val.name)}
                            />
                        </div>
                    ))}
                </Carousel>
            </WingBlank>
        </div>
    )
}