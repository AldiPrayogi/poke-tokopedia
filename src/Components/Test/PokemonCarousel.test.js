import {shallow} from 'enzyme';
import {PokemonCarousel} from "../PokemonCarousel";
import {PokemonDataCarousel} from "../../Fixtures/PokemonData";

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush,
    }),
}));

afterEach(() =>{
    jest.clearAllMocks();
})

afterAll(() => {
    jest.clearAllMocks();
})

describe('PokemonCarousel', () => {
    const wrapper = shallow(<PokemonCarousel data={PokemonDataCarousel}/>);
    
    describe('render', () => {

        it('should render correctly with data', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
    describe('handlePokemonClick', () => {
        it('should call history push using pokemonName when pokemon is clicked', () => {
            const firstPokemon = wrapper.find('.pokemon-image').at(0);

            firstPokemon.simulate('click');

            expect(mockHistoryPush).toHaveBeenCalledWith({'pathname': `/pokemon-detail/${PokemonDataCarousel[0].name}`});
        });
    });
});