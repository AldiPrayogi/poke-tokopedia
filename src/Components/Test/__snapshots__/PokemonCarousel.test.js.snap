// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCarousel render should render correctly with data 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="PokemonCarousel"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "map": undefined,
      "name": "djvarp",
      "next": undefined,
      "styles": "
    width: 100%;
    text-align: center
    .carousel-container{
        textAlign: center
    }
    .pokemon-name-header{
        background-color: #222224;
        width: 110%
    }
    .pokemon-name{
        color: #f0c44d;
        font-size: 2.7vh;
        width: 100%;
        padding-right: 9%;
    }
    .am-carousel-wrap{
        margin-top: 100px important!;
    }
    img{
        height: 25vh;
        vertical-align: top;
        padding-right:2%;
    }
",
      "toString": [Function],
    }
  }
>
  <WingBlank
    prefixCls="am-wingblank"
    size="lg"
  >
    <Carousel
      arrows={false}
      autoplay={false}
      cellAlign="center"
      cellSpacing={10}
      dotActiveStyle={Object {}}
      dotStyle={Object {}}
      dots={true}
      infinite={true}
      prefixCls="am-carousel"
      selectedIndex={0}
      slideWidth={1}
    >
      <div
        className="carousel-container"
      >
        <div
          className="pokemon-name-header"
        >
          <h4
            className="pokemon-name"
          >
            IVYSAUR
          </h4>
        </div>
        <WhiteSpace
          prefixCls="am-whitespace"
          size="xl"
        />
        <img
          alt="ivysaur"
          className="pokemon-image"
          onClick={[Function]}
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
        />
      </div>
      <div
        className="carousel-container"
      >
        <div
          className="pokemon-name-header"
        >
          <h4
            className="pokemon-name"
          >
            IVYSAUR
          </h4>
        </div>
        <WhiteSpace
          prefixCls="am-whitespace"
          size="xl"
        />
        <img
          alt="ivysaur"
          className="pokemon-image"
          onClick={[Function]}
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
        />
      </div>
      <div
        className="carousel-container"
      >
        <div
          className="pokemon-name-header"
        >
          <h4
            className="pokemon-name"
          >
            IVYSAUR
          </h4>
        </div>
        <WhiteSpace
          prefixCls="am-whitespace"
          size="xl"
        />
        <img
          alt="ivysaur"
          className="pokemon-image"
          onClick={[Function]}
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
        />
      </div>
    </Carousel>
  </WingBlank>
</EmotionCssPropInternal>
`;
